zipline/finance/trading.py

```diff
@@ -93,13 +92,34 @@ def __init__(
         max_date=None,
         env_trading_calendar=tradingcalendar
     ):
+        # `tc_tdays` is short for "trading calendar trading days"
+        # `tc_tday` is short for "trading calendar trading day"
+        tc_tdays = env_trading_calendar.trading_days
+        tc_tday = env_trading_calendar.trading_day
+
+        if max_date:
+            self.trading_days = tc_tdays[tc_tdays <= max_date].copy()
+            self.trading_day = tc_tday[tc_tday <= max_date].copy()
```

Comment:
This throws an error for me:

 File ".../zipline/finance/trading.py", line 102, in __init__
    self.trading_day = tc_tday[tc_tday <= max_date].copy()
TypeError: can't compare datetime.datetime to CustomBusinessDay